### PR TITLE
Feature/results reports

### PR DESCRIPTION
**Pull request by yurii-krupa**
Add functionality to show Test(a.k.a Quiz) results by group.
Add report printing to .pdf file and sending it via email

### CODE REVIEW

**Comment by yurii-krupa** [comment]
[Results](https://trello.com/c/fOT1dZjp/25-results)

File: DTAdmin/Group/GroupDetailsViewController.swift [hunk 1]
```diff
@@ -24,6 +24,15 @@ class GroupDetailsViewController: UIViewController {
     @IBAction func getTimeTableByGroupTapped(_ sender: Any) {
     }
     @IBAction func getResultsByGroupTapped(_ sender: Any) {
+      guard let resultByGroupViewController = UIStoryboard(name: "Result", bundle: nil).instantiateViewController(withIdentifier: "ResultByGroup") as? ResultByGroupViewController else  { return }
+      
+//      guard let test = tests?[indexPath.row] else {
```
Comment (iostaf) [inline DTAdmin/Group/GroupDetailsViewController.swift:29]: Please remove commented dead code or provide good justification why have you commented it.

File: DTAdmin/Models/DataManager.swift [hunk 2]
```diff
@@ -583,6 +583,153 @@ class DataManager: HTTPManager {
...
+    guard let url = URL(string: httpManager.urlProtocol + httpManager.urlDomain + "/Result/GetRecordsByTestGroupDate/" + test["id"]! + "/" + group["id"]!) else { return }
+    
+    var request = URLRequest(url: url)
+    request.httpMethod = "GET"
+    request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+    request.setValue("utf-8", forHTTPHeaderField: "Charset")
+    request.setValue(cookie[Keys.cookie], forHTTPHeaderField: Keys.cookie)
+    
+    _ = URLSession.shared.dataTask(with: request) { (data, response, error) in
+      if let error = error {
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift:600]: Please use 4 spaces for indentation

File: DTAdmin/Models/DataManager.swift [hunk 3]
```diff
@@ -583,6 +583,153 @@ class DataManager: HTTPManager {
...
+        completionHandler(error.localizedDescription, nil)
+      } else {
+        if let sessionResponse = response as? HTTPURLResponse, let sessionData = data {
+          if sessionResponse.statusCode == 200 {
```
Comment (iostaf) [inline DTAdmin/Models/DataManager.swift:604]: Why do you use 200 here? AFAIK we have an enum for response codes.

File: DTAdmin/Models/Structures/ResultStructure.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+//  ResultStructure.swift
+//  DTAdmin
+//
+//  Created by Yurii Krupa on 11/7/17.
+//  Copyright © 2017 if-ios-077. All rights reserved.
+//
+
+import Foundation
+
+struct ResultStructure: Serializable {
```
Comment (iostaf) [inline DTAdmin/Models/Structures/ResultStructure.swift:11]: Please rename `ResultStructure` to something more meaningful and not use Structure suffix

**Review by iostaf — CHANGES_REQUESTED** [review]

**Pull request closed without merging**